### PR TITLE
Fixed remaining MyPy problems

### PR DESCRIPTION
**Pull request by pkubanek**
- Enable MyPy checks
- MyPy fixes

### CODE REVIEW

**Review by couger01 — APPROVED** [review]
LGTM but remove setup.cfg as ts_pre_commit_conf makes it unnecessary.

**Review by teweitsai — APPROVED** [review]
Good to me. For the data from SAL, you can use the following annotation:
https://github.com/lsst-ts/ts_salobj/blob/develop/python/lsst/ts/salobj/type_hints.py#L37

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -19,4 +19,4 @@ jobs:
         run: pip install -r <(curl https://raw.githubusercontent.com/lsst/linting/master/requirements.txt)
 
       - name: Run linter
-        run: flake8
+        run: flake8 --max-line-length 120 --exclude __init__.py --ignore E203,W503
```
Comment (teweitsai) [inline .github/workflows/lint.yaml:22]: I think you could use our default setting of `.flake8` by `.ts_pre_commit_config.yaml`.

**Review by teweitsai — APPROVED** [review]
It looks good to me. To be honest, this PR has > 5000 lines code change and my browser could not open it well to review the code. Therefore, I only check the setting files. Thanks!

**Review by couger01 — APPROVED** [review]
Can't really review all of the MYPY changes, but LGTM otherwise. I suggest removing the GH action as it's redundant with Jenkinsfile usage.

**Pull request merged**